### PR TITLE
Enable uploading of QR Codes to the Gini API

### PR DESCRIPTION
**Pull request by a-szotyori**
To track QR Codes we had to change our beta QR Code public api quite a bit. We don't expose the payment data anymore directly. We create a json containing the raw QR Code string and the payment data we parsed from it (#135). This json is publicly available as an utf-8 encoded byte array which should be uploaded to the Gini API to get the payment data from the QR Code. Feedback should also be sent for it.

The example apps were updated to upload the QR Code. Documentation was also updated.

**Note:**
The previous beta QR Code public api used the term `Payment Data` to cover future cases where other sources than QR Codes are used to read payment data (e.g. NFC). The requirement to upload the QR Code required us to drop that naming and mention QR Codes explicitly for clarity and convenience. That's why there are so many changed files.

### How to test
Run the example apps and scan BezahlCode, Stuzza or GiroCode.

### Ticket 
Issue #137

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentInterface.java [hunk 1]
```diff
@@ -79,4 +81,29 @@
...
+    /**
+     * <p>
+     *     Call this method to hide the activity indicator and enable user interaction.
+     * </p>
+     */
+    void hideActivityIndicatorAndEnableInteraction();
+
+    /**
+     * <p>
+     *     Call this method when to show an error message to the user in the Camera Screen.
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentInterface.java]: ~when~

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java [hunk 2]
```diff
@@ -297,7 +297,8 @@
  * </ul>
  * </p>
  **/
-public class CameraActivity extends AppCompatActivity implements CameraFragmentListener {
+public class CameraActivity extends AppCompatActivity implements CameraFragmentListener,
+        CameraFragmentInterface {
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java]: I'm not really sure about this interface name. When I have seen it, I have thought about a generic java `interface`, not as an interface related to user interface.

File: screenapiexample/src/main/java/net/gini/android/vision/screen/CameraScreenApiActivity.java [hunk 3]
```diff
@@ -161,12 +174,27 @@ private boolean isRenderablePDF(final Uri uri) {
...
-        finish();
+    public void onQRCodeAvailable(@NonNull final QRCodeDocument qrCodeDocument) {
+        showActivityIndicatorAndDisableInteraction();
+        mSingleDocumentAnalyzer.cancelAnalysis();
+        mSingleDocumentAnalyzer.analyzeDocument(qrCodeDocument,
+                new DocumentAnalyzer.Listener() {
+                    @Override
+                    public void onException(final Exception exception) {
+                        hideActivityIndicatorAndEnableInteraction();
+                        showError("Could not use the QR Code. Try again or take a picture of your document.", 4000);
```
Comment (kikettas) [inline screenapiexample/src/main/java/net/gini/android/vision/screen/CameraScreenApiActivity.java]: Same here.

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java (hunk 2 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java]: It is a generic interface and it contains only UI related methods now which might change in the future.

**Pull request merged**